### PR TITLE
fix integration tests in docker

### PR DESCRIPTION
**Pull request by runcom**
Trying to tackle performance issues as found in #324 
Also, fixing a bunch of stuff to enable network and seccomp tests in container (otherwise disabled...).
I can run `make integration` in about ~7/8min with this branch.

TODO:

- [x] enable integration tests in Travis
- [x] cleanup commits (split the one there's right now)

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
seems like image tests ran on Travis :confused: and they passed as well

**Comment by runcom** [comment]
let's see how it goes with all tests enabled :smiling_imp:

**Comment by runcom** [comment]
Awesome all tests run and pass on Travis! :tada: I'm going to re-push with a clean commit history and let this run again (hopefully it'll be green again). @mrunalp @jawnsy PTAL

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.7.3
+FROM golang:1.7.4
```
Comment (jawnsy) [inline Dockerfile]: Consider using `golang:1.7` instead, which I think would just pull the latest go 1.7.x image

**Comment by jawnsy** [comment]
LGTM, will give this a spin on my local system as well :+1:

**Comment by runcom** [comment]
> LGTM, will give this a spin on my local system as well

Thanks @jawnsy I'm not sure this would bring down you test-time-performance issue. Worth a try btw. This is mainly enabling tests in Travis (finally!)

**Comment by mrunalp** [comment]
Sweet! I am running this locally now.

**Comment by mrunalp** [comment]
I see this error:
```
make[1]: 'copyimg' is up to date.
make[1]: Leaving directory '/go/src/github.com/kubernetes-incubator/cri-o/test/copyimg'
./test/copyimg/copyimg: error while loading shared libraries: libdevmapper.so.1.02: cannot open shared object file: No such file or directory
The command '/bin/sh -c make copyimg    && mkdir -p .artifacts/redis-image      && ./test/copyimg/copyimg --import-from=docker://redis --export-to=dir:.artifacts/redis-image --signature-policy ./test/policy.json' returned a non-zero code: 127
make: *** [Makefile:91: ocidimage] Error 127
```

**Comment by runcom** [comment]
@mrunalp fixed it, I had wrongly removed a `start_ocid` call.

**Comment by mrunalp** [comment]
One test is failing for me:
```
not ok 8 ctrs status for a pod
# (in test file ./ctr.bats, line 106)
#   `[ "$status" -eq 0 ]' failed
# 2017/01/19 18:48:47 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial unix /tmp/tmp.BDU9olnHNk/ocid.sock: connect: no such file or directory"; Reconnecting to {"/tmp/tmp.BDU9olnHNk/ocid.sock" <nil>}
# Creating the pod sandbox failed: rpc error: code = 14 desc = grpc: the connection is unavailable
```

**Comment by runcom** [comment]
@mrunalp please resync your local copy of this PR I fixed that in the last commit

**Comment by runcom** [comment]
tests are green on Travis also :tada:

**Comment by mrunalp** [comment]
@runcom okay, retesting.

**Comment by mrunalp** [comment]
okay, ran in 12 minutes for me (in single user mode though). LGTM.

**Pull request merged**